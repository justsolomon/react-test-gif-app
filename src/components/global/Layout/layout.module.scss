@import "../../../assets/scss/base/variables";

.layout {
  &__background {
    height: 200px;
    width: 100%;
    display: flex;
    align-items: center;
    justify-content: center;
    background-color: $bg-light--tertiary;

    form {
      width: 70%;

      @media #{$media-tablet} {
        width: 85%;
      }
    }

    @media #{$media-mobile-sm} {
      height: 150px;

      form {
        width: 90%;
      }
    }
  }

  &__content {
    margin: 1rem 7rem;
    padding-bottom: 1rem;

    @media #{$media-desktop-sm} {
      margin: 1rem 4rem;
    }

    @media #{$media-tablet} {
      margin: 1rem 2rem;
    }

    @media #{$media-mobile} {
      margin: 1rem;
    }
  }
}
