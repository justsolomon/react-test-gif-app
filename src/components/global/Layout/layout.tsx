import SearchFieldContainer from "containers/SearchFieldContainer";
import styles from "./layout.module.scss";

interface ILayout {
  children?: React.ReactNode;
}

const Layout = ({ children }: ILayout): JSX.Element => {
  return (
    <div className={styles["layout"]}>
      <div className={styles["layout__background"]}>
        <SearchFieldContainer />
      </div>
      <div className={styles["layout__content"]}>{children}</div>
    </div>
  );
};

export default Layout;
