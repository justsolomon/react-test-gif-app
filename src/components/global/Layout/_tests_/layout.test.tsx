import { render } from "@testing-library/react";
import Layout from "../layout";

jest.mock("react-router", () => ({
  ...jest.requireActual("react-router"),
  useLocation: () => ({
    state: null
  })
}));

describe("<Layout />", () => {
  it("should render the layout component properly", () => {
    const { asFragment } = render(<Layout>Test Layout</Layout>);

    expect(asFragment()).toMatchSnapshot();
  });
});
