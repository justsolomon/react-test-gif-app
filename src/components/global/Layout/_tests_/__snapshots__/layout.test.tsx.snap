// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Layout /> should render the layout component properly 1`] = `
<DocumentFragment>
  <div
    class="layout"
  >
    <div
      class="layout__background"
    >
      <form>
        <div
          class="search-input__container"
        >
          <input
            class="search-input__field"
            placeholder="Search for gif"
            value=""
          />
          <button
            class="button  button--solid button--none "
            type="submit"
          >
            <svg>
              search.svg
            </svg>
          </button>
        </div>
      </form>
    </div>
    <div
      class="layout__content"
    >
      Test Layout
    </div>
  </div>
</DocumentFragment>
`;
